sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java

```diff
@@ -3070,8 +3147,8 @@ public Mono<OpenAIFile> getFile(String fileId) {
      * @return represents an assistant that can call the model and use tools on successful completion of {@link Mono}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    Mono<OpenAIFile> uploadFile(FileDetails file, FilePurpose purpose, String filename) {
-        // Generated convenience method for uploadFileWithResponse
+    public Mono<OpenAIFile> uploadFile(FileDetails file, FilePurpose purpose, String filename) {
+        file.setFilename(filename);
```

Comment:
I wrote it like this so that `String filename` takes precedence over what is passed in the `FileDetails` options, since in the case of OpenAI `filename` needs to be defined for the multipart request.

Reply:
This issue will be addressed in a separate PR where we will fix this problem which is an upstream issue with the spec (`filename` shouldn't be there to begin with)